gp2-stress-experiment.ipynb

```diff
@@ -0,0 +1,933 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "num_spectra = stitched.coords[\"spectrum\"].shape[0]\n",
+    "bank_width = int(np.sqrt(num_spectra))\n",
+    "integrated = operations.integrate(stitched)\n",
+    "integrated_neighbour_spectra = operations.generate_neighbouring_spectra(num_spectra=num_spectra,\n",
+    "                                                                        bank_width=bank_width)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def mask_non_sample_regions(integrated_dataset, var_name):\n",
+    "    integrated_spectra = integrated_dataset[var_name]\n",
+    "    spectra_masks = [False] * num_spectra\n",
+    "    for i, spec_val in enumerate(integrated_spectra.values):\n",
+    "        if spec_val > masking_threshold:\n",
+    "            spectra_masks[i] =  True\n",
+    "\n",
+    "    for i, adj_spectra in integrated_neighbour_spectra.items():\n",
+    "        if all(spectra_masks[i] == True for i in adj_spectra):\n",
+    "            spectra_masks[i] = True  # Force to true as neighbours true\n",
+    "\n",
+    "        if all(spectra_masks[i] == False for i in adj_spectra):\n",
+    "            spectra_masks[i] = False\n",
+    "\n",
+    "    stitched.masks[var_name] = sc.Variable([\"spectrum\"], values=spectra_masks)\n",
+    "\n",
+    "for k in integrated.keys():\n",
+    "    if k == \"reference\":\n",
+    "        continue\n",
+    "\n",
+    "    print(f\"Masking non-sample regions in {k}\")\n",
+    "    integrated[k] = integrated[k] / integrated[\"reference\"]\n",
+    "    mask_non_sample_regions(integrated, k)\n",
+    "\n",
+    "    if output_tiff_stack:\n",
+    "        print(f\"Exporting tiff stack for {k}\")\n",
+    "        tiff_out_dir = os.path.join(output_dir, f\"tiffs_tof_sum_{k}\")\n",
+    "        imaging.export_tiff_stack(dataset=integrated, key=k,\n",
+    "                                  base_name=f\"{k}_norm_sum\",\n",
+    "                                  output_dir=tiff_out_dir,\n",
+    "                                  x_len=324, y_len=324)\n",
+    "        # Print a newline to prevent the saving messages overlapping\n",
+    "        print()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Normalize by open beam\n",
+    "normalized = stitched.copy()\n",
+    "normalized[\"sample\"] = stitched[\"sample\"] / stitched[\"reference\"]\n",
+    "normalized[\"sample_elastic\"] = stitched[\"sample_elastic\"] / stitched[\"reference\"]\n",
+    "normalized[\"sample_plastic\"] = stitched[\"sample_plastic\"] / stitched[\"reference\"]\n",
```

Comment:
I have now seen more than one occassion indicating that having `reference` in the same dataset hurts you (requireing very verbose code). Consider making it separate, then this reduces to

```python
normalized = stitched / reference
```